consensus/propagation/have_wants.go

```diff
@@ -69,11 +73,12 @@ func (blockProp *Reactor) handleHaves(peer p2p.ID, haves *proptypes.HaveParts) {
 	for _, index := range hc.GetTrueIndices() {
 		select {
 		case <-blockProp.ctx.Done():
-		case p.requestChan <- request{
+		case p.receivedHaves <- request{
 			height: height,
 			round:  round,
 			index:  uint32(index),
 		}:
+			p.RequestsReady()
```

Comment:
adding another channel allows us to remove any timeouts for waiting w/o having to deal with the output of the `receivedHaves` channel if we're not ready to send stuff

Reply:
if I understand correctly, the `RequestReady()` is supposed to transmit that we finished queuing individual requests? if so, then it should be called after the loop. Also, what if the number of requests > peer capactity, will this block?

Reply:
nah check the function

Reply:
so, is this supposed to just notify that there is a new request coming? seems redundant to the channel message. Or is it for: 

> the `RequestReady()` is supposed to transmit that we finished queuing individual requests?